### PR TITLE
Containerservices agentpoolprofile maxpods

### LINKED ISSUE #1729: AKS Agent Pool Profile missing maxPods
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version
Terraform v0.11.7
+ provider.azurerm v1.12.0
+ provider.random v1.3.1

### Affected Resource(s)

* azurerm_kubernetes_cluster

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster" "aks_container" {
  name       = "akc-${random_integer.random_int.result}"
  location   = "${var.resource_group_location}"
  dns_prefix = "akc-${random_integer.random_int.result}"

  resource_group_name = "${azurerm_resource_group.akc-rg.name}"

  linux_profile {
    admin_username = "${var.linux_admin_username}"

    ssh_key {
      key_data = "${var.linux_admin_ssh_publickey}"
    }
  }

  agent_pool_profile {
    name    = "agentpool"
    count   = "2"
    vm_size = "Standard_DS2_v2"
    os_type = "Linux"
    maxPods = "90"

    # Required for advanced networking
    vnet_subnet_id = "${azurerm_subnet.aks_subnet.id}"
  }

  service_principal {
    client_id       = "${var.client_id}"
    client_secret   = "${var.client_secret}"
  }

  network_profile {
    network_plugin     = "azure"
    dns_service_ip     = "10.0.0.10"
    docker_bridge_cidr = "172.17.0.1/16"
    service_cidr       = "10.0.0.0/16"
  }
}
```

### Expected Behavior

I should be able to declare a value for 'maxPods' in the agent_pool_profile based on microsoft documentation: https://docs.microsoft.com/en-us/azure/aks/networking-overview#configure-maximum

### Actual Behavior

This parameter is not recognized by terraform. Doc link: https://www.terraform.io/docs/providers/azurerm/r/kubernetes_cluster.html#agent_pool_profile

Debug output: ```Error: azurerm_kubernetes_cluster.aks_container: agent_pool_profile.0: invalid or unknown key: maxPods```

### Steps to Reproduce

N/A

### Important Factoids

N/A

### References

N/A

* #0000

### PR DESCRIPTION
**Pull request by lfshr**
Added max_pods to agent_pool_profile in azurerm_kubernetes_cluster. This fixes #1729

### CODE REVIEW

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @lfshr 

Thanks for this PR :)

I've left a comment inline about the default value - but this otherwise LGTM; if we're able to confirm that and add a test for this (to ensure the `max_pods` value is set as expected) then this otherwise LGTM 👍 

Thanks!

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
~removing duplicate comment~

File: azurerm/resource_arm_kubernetes_cluster.go
Comment (lfshr) [reply azurerm/resource_arm_kubernetes_cluster.go]: Of course it does... Doh!

Comment (lfshr) [reply azurerm/resource_arm_kubernetes_cluster.go]: Fixed via 8a40400
In my defense, I'm still in holiday mode! :stuck_out_tongue_closed_eyes: 

Thanks Tom!

Comment (lfshr) [reply azurerm/resource_arm_kubernetes_cluster.go]: Just going to check the fact it's an int isn't going to cause problems. I may need to conditionally supply this to the SDK (if value > 0)

Comment (lfshr) [reply azurerm/resource_arm_kubernetes_cluster.go]: Hold off on this for now. I can't seem to login to the az client this morning. @tombuildsstuff could you possibly try and spin up a cluster with no max_pods specified and tell me if it works? :confused:

Comment (tombuildsstuff) [reply azurerm/resource_arm_kubernetes_cluster.go]: @lfshr 

> In my defense, I'm still in holiday mode! 😝
heh no worries, thanks for fixing it - hope you enjoyed it :)

I've spun one up using the existing resource (which sets nothing) - which defaults this to `110` - so I think this should be fine 👍

File: azurerm/resource_arm_kubernetes_cluster.go
Comment (tombuildsstuff) [inline azurerm/resource_arm_kubernetes_cluster.go]: could we update the acceptance tests (for the internal network probably makes the most sense, rather than a new test?) to set this value?

File: website/docs/r/kubernetes_cluster.html.markdown
Comment (tombuildsstuff) [inline website/docs/r/kubernetes_cluster.html.markdown]: This no longer defaults to 30 - so can we remove this?

**Review by tombuildsstuff — APPROVED** [review]
Hey @lfshr 

Thanks for pushing those changes - if we can update the acceptance tests to check this functionality then this otherwise LGTM 👍 

Thanks!

File: azurerm/resource_arm_kubernetes_cluster.go
Comment (lfshr) [reply azurerm/resource_arm_kubernetes_cluster.go]: Done :)

**Comment by lfshr** [comment]
@tombuildsstuff That should be everything :)

**Review by tombuildsstuff — APPROVED** [review]
hey @lfshr 

Thanks for pushing the latest changes - this now LGTM 👍 - I'll kick off the tests shortly :)

Thanks!

**Comment by tombuildsstuff** [comment]
Tests pass - thanks again for this @lfshr :)

screenshot 2018-08-14 at 13 42 25

**Comment by yves-vogl** [comment]
Thank you

**Pull request merged**
